test/nodes/test_retriever.py

```diff
@@ -648,6 +660,58 @@ def test_table_text_retriever_training(tmp_path, document_store, samples_path):
     )
 
 
+@pytest.mark.integration
+def test_sentence_transformers_retriever_training():
+    retriever = EmbeddingRetriever(
+        embedding_model="sentence-transformers/all-MiniLM-L6-v2", model_format="sentence_transformers", use_gpu=False
+    )
+    retriever.train(
+        training_data=[
+            {
+                "question": "What color is the sky?",
+                "pos_doc": "The color of the sky is blue.",
+                "neg_doc": "The color of the grass is brown.",
+            },
+            {
+                "question": "What is the capital of Germany?",
+                "pos_doc": "The capital of Germany is Berlin.",
+                "neg_doc": "The capital of France is Paris.",
+            },
+        ],
+        train_loss="mnrl",
+        n_epochs=1,
+        batch_size=1,
+        num_warmup_steps=0,
+    )
+
+
+@pytest.mark.integration
+def test_sentence_transformers_retriever_training_with_gradient_checkpointing():
+    retriever = EmbeddingRetriever(
+        embedding_model="sentence-transformers/all-MiniLM-L6-v2", model_format="sentence_transformers", use_gpu=False
+    )
+    retriever.train(
+        training_data=[
+            {
+                "question": "What color is the sky?",
+                "pos_doc": "The color of the sky is blue.",
+                "neg_doc": "The color of the grass is brown.",
+            },
+            {
+                "question": "What is the capital of Germany?",
+                "pos_doc": "The capital of Germany is Berlin.",
+                "neg_doc": "The capital of France is Paris.",
+            },
+        ],
+        train_loss="mnrl",
+        n_epochs=1,
+        batch_size=1,
+        num_warmup_steps=0,
+        gradient_checkpointing=True,
+    )
```

Comment:
Sounds good, I've updated the test to use mocking to skip the training step to avoid calling the train code in sentence transformers. I didn't see a great way to assert anything in the test since the train method of EmbeddingRetriever doesn't return anything. So I check that the test itself runs, which tests all Haystack code up until the call to the fit method. 

Please let me know if this is the type of mocking you were thinking about.

Reply:
90% there! I'll leave another comment on the new version